crates/mun_hir/src/ty/infer.rs

```diff
@@ -400,9 +400,23 @@ impl<'a, D: HirDatabase> InferenceResultBuilder<'a, D> {
                     }
                 }
             }
-            Expr::UnaryOp { .. } => Ty::Unknown,
-            //            Expr::UnaryOp { expr: _, op: _ } => {}
-            //            Expr::Block { statements: _, tail: _ } => {}
+            Expr::UnaryOp { expr, op } => {
+                let ty = self.infer_expr(*expr, &Expectation::none());
+                if let Some(simple) = ty.as_simple() {
+                    match op {
+                        UnaryOp::Not => match simple {
+                            TypeCtor::Bool | TypeCtor::Int(_) => ty,
+                            _ => Ty::Unknown,
+                        },
+                        UnaryOp::Neg => match simple {
+                            TypeCtor::Float(_) | TypeCtor::Int(_) => ty,
+                            _ => Ty::Unknown,
```

Comment:
ditto

Reply:
?

Reply:
Same as the above comment:

There should be an error emitted here. Something similar to binary operation error: cannot apply operator `-` on type `int`.

Reply:
Okay.